src/Test.sol

```diff
@@ -448,8 +449,471 @@ abstract contract Test is DSTest, Script {
+        artifact.transactionReturns = rawArtifact.transactionReturns;
+        artifact.receipts = rawToConvertedReceipts(rawArtifact.receipts);
+        artifact.transactions = rawToConvertedEIP1559Txs(rawArtifact.transactions);
+        return artifact;
+    }
+
+    // break into two functions
+    function rawToConvertedEIP1559Txs(RawEIP1559Transaction[] memory rawTxs)
+        internal
+        returns (EIP1559Transaction[] memory)
+    {
+        EIP1559Transaction[] memory txs = new EIP1559Transaction[](rawTxs.length);
+        for(uint i;i<rawTxs.length;i++){
+            txs[i] = rawToConvertedEIP1559Tx(rawTxs[i]);
+        }
+        return txs;
+    }
+
+    function rawToConvertedEIP1559Tx(RawEIP1559Transaction memory rawTx)
+        internal
+        returns (EIP1559Transaction memory)
+    {
+            EIP1559Transaction memory transaction;
+            transaction.arguments = rawTx.arguments;
+            transaction.contractName = rawTx.contractName;
+            transaction.functionName = rawTx.functionName;
+            transaction.hash= rawTx.hash;
+            transaction.detail = rawToConvertedEIP1559Detail(rawTx.detail);
+    }
+
+    function rawToConvertedEIP1559Detail(RawEIP1559TransactionDetail memory rawDetail)
+        internal
+        returns (EIP1559TransactionDetail memory)
+    {
+        EIP1559TransactionDetail memory detail;
+        detail.data = rawDetail.data;
+        detail.from = rawDetail.from;
+        detail.to = rawDetail.to;
+        detail.nonce = bytesToUint(rawDetail.nonce);
+        detail.txType = bytesToUint(rawDetail.txType);
+        detail.value = bytesToUint(rawDetail.value);
+        detail.gas = bytesToUint(rawDetail.gas);
+        detail.accessList = rawDetail.accessList;
+        return detail;
+
+    }
+
+
+    // Read in all EIP1559 deployments transactions.
+    function readEIP1559Transactions(string memory path)
+        internal
+        returns (EIP1559Transaction[] memory)
+    {
+        string memory deployData = vm.readFile(path);
+        bytes memory parsedDeployData =
+            vm.parseJson(deployData, ".transactions");
+        RawEIP1559Transaction[] memory rawTxs = abi.decode(parsedDeployData, (RawEIP1559Transaction[]));
+        return rawToConvertedEIP1559Txs(rawTxs);
+    }
+
+
+    function readEIP1559Transaction(string memory path, uint256 index)
+        internal
+        returns (EIP1559Transaction memory)
+    {
+        string memory deployData = vm.readFile(path);
+        string memory key = string(abi.encodePacked(".transactions[",vm.toString(index), "]"));
+        bytes memory parsedDeployData =
+            vm.parseJson(deployData, key);
+        RawEIP1559Transaction memory rawTx = abi.decode(parsedDeployData, (RawEIP1559Transaction));
+        return rawToConvertedEIP1559Tx(rawTx);
+    }
+
+
+    // Analogous to readTransactions, but for receipts.
+    // It requires --via-ir to avoid `stack too deep errors`
```

Comment:
This comment is now outdated right?

Reply:
Saw you marked this one resolved but the comment is still there, so just want to follow up